Resources/public/Admin.js

```diff
@@ -139,47 +138,11 @@ var Admin = {
     },
 
     /**
-     * display related errors messages
-     *
-     * @param subject
+     * @deprecated in version 2.4
      */
-    add_pretty_errors: function(subject) {
-        Admin.log('[core|add_pretty_errors] configure pretty errors on', subject);
-        jQuery('div.sonata-ba-field-error', subject).each(function(index, element) {
-            var input = jQuery(':input', element);
-
-            if (!input.length) {
-                return;
-            }
-
-            var message = jQuery('div.sonata-ba-field-error-messages', element).html();
-            jQuery('div.sonata-ba-field-error-messages', element).remove();
-
-            if (!message || message.length == 0) {
-                return;
-            }
-
-            var target = input,
-                fieldShortDescription = input.closest('.field-container').find('.field-short-description'),
-                select2 = input.closest('.select2-container')
-                ;
-
-            if (fieldShortDescription.length) {
-                target = fieldShortDescription;
-            } else if (select2.length) {
-                target= select2;
-            }
-
-            target.popover({
-                content: message,
-                trigger: 'hover',
-                html: true,
-                placement: 'top',
-                template: '<div class="popover"><div class="arrow"></div><div class="popover-inner"><div class="popover-content alert-error"><p></p></div></div></div>'
-            });
-
-        });
-    },
+    add_pretty_errors: function() {
+        console.warn('Admin.add_pretty_errors() was deprecated in version 2.4');
```

Comment:
It depends.
You can't throw warnings in JS, so an uncatched exception will most likely break your program.
The current trend is to write:
```js
if (process.env.NODE_ENV !== 'production') console.warn('foo') // or throw new Whatever()
// or
if (__DEV__) throw new Error('wtf?') // or console.error('...')
```
And then have a build step with [envify](https://github.com/hughsk/envify) or webpack's [DefinePlugin](https://webpack.github.io/docs/list-of-plugins.html#defineplugin) , to allow UglifyJS to remove these statements in production builds.
Unfortunately, while using gulp/webpack is relatively easy at the application level, I'm not aware of a good integration layer for bundles, (like [elixir](https://github.com/laravel/elixir) in laravel). [Puli](http://docs.puli.io/en/latest/) looks really interesting, but I havn't had the time to really look into it.

Anyway, since this is not a critical codepath, I thought that turning it to a noop a warn the developper would be more friendly.

In the end it's a matter of deciding what would be a good deprecation system for the client-side layer.